subsys/net/lib/sockets/sockets.c

```diff
@@ -257,8 +276,14 @@ int z_impl_zsock_close(int sock)
 
 	NET_DBG("close: ctx=%p, fd=%d", ctx, sock);
 
-	return z_fdtable_call_ioctl((const struct fd_op_vtable *)vtable,
-				    ctx, ZFD_IOCTL_CLOSE);
+	(void)sys_mutex_lock(lock, K_FOREVER);
```

Comment:
The mutex is a part of `struct fd_entry`, which got released just before. I think we should move `z_free_fd(sock);` after the mutex operation.

Reply:
Good point, fixed.